### ISSUE TITLE
O projeto está morto ?

### ISSUE THREAD

**Issue by weslenbreno** [bti-imd/IntegraTI-API#39]
Ainda existe a possibilidade desse porjeto ir pra frente ou vamos abandonar completamente ?

**Comment by rodrigondec** [comment]
Acredito que devido à falta de tempo/comprometimento do pessoal ele deve ter morrido aos poucos. Com as disciplinas para me formar também não pude dar tanta atenção. 

Agora que estou formado tenho mais conhecimento na área e estou disponível se quiserem voltar a ativa.

Como está a situação para você, @Mazuh ?

**Comment by Mazuh** [comment]
Estou sempre programando, dessa vez mil vezes melhor, e ainda disposto a colaborar com free source. ^_^ Porém não tenho o mesmo tempo de antes pra coletar requisitos e manter o sistema ativo com dados, acompanhamento de usuários e tudo mais. Ou seja, mesmo que a gente desenvolva, não consigo enxergar como ele será bem aproveitado.

**Comment by rodrigondec** [comment]
Sim, o calo desse projeto/organização sempre foi não ter alguem que é 'contratado' pra ser resposável por ele. Na minha opnião seria algo pros integrantes do centro acadêmico do curso ou algo como uma bolsa de apoio técnico. @gellyviana como ficou o centro acadêmico?

**Comment by rodrigondec** [comment]
Sim, o calo desse projeto/organização sempre foi não ter alguem que é 'contratado' pra ser resposável por ele. Na minha opnião seria algo pros integrantes do centro acadêmico do curso ou algo como uma bolsa de apoio técnico. @gellyviana como ficou o centro acadêmico?

**Comment by gellyviana** [comment]
Olá pessoal, me desculpem pela demora na resposta, mas estou em semana de prova e em paralelo tive que buscar e trazer para aqui soluções para a continuação do projeto. Respondendo a pergunta de @rodrigondec, o Diretório Academico Ada Lovelace (DAAL) finalmente foi fundado no BTI e temos menos burocracias para conseguir algumas coisas, o que cansa mais no meu ponto de vista. Confesso que ultimamente tenho pensado muito no nosso projeto e vejo o quanto é pertinente deixar um legado para o nosso curso. Em breve trarei mais novidades e conto com a ajuda de vocês .

**Comment by weslenbreno** [comment]
Show de bola, pode contar comigo.

**Comment by gellyviana** [comment]
Pessoal, estive conversando com Lucas, responsável pelo repositório do DAAL  e tenho duas coisas para dizer:

1º Quero saber se haveria possibidade de mudar a atual tecnologia que estavamos trabalhando para DJANGO, porque dessa forma oportunizaria um grande grupo(Grupy, Puladies e os alunos do BTI) contribuir com o repositório;
2º Com a aceitação da migração, será criado um projeto no repositório do DAAL;

É claro que muita coisa que atualmente estavamos fazendo seria passado para esse novo projeto, acredito que também que a parte de front-end poderia ser aproveitada, se estiver errada me corrija @weslenbreno. Mas também quero ouvir sugestões de vocês. Aguardo a retorno de vocês e até!

**Comment by rodrigondec** [comment]
@gellyviana , sobre essa primeira, eu já fiz a branch [django](https://github.com/discentes-imd/IntegraTI-API/tree/django) em que estava migrando o projeto pra Django. Foi discutido com @weslenbreno e o resto do pessoal que estava envolvido no back.

**Comment by gellyviana** [comment]
Bem lembrado, então adicionaria o nosso repositório no conta do DAAL ou criaria um novo repositório no DAAL mas foco em Django? . Eu acho que seria mais interessante fazermos um novo repositório no DAAL usando a nova tecnologia.

**Comment by rodrigondec** [comment]
Sim, estava tão no incio que só aproveita a modelagem que fizemos. Mas temos no repo os documentos

**Comment by weslenbreno** [comment]
Acho mais produtivo fazer em Django mesmo, tem muito mais gente que manja e é um framework incrível, atende perfeitamente as nossas necessidades.

@gellyviana sobre o front, acredito que seja mais fácil refazer, nós não tínhamos ainda tanta coisa, houveram algumas mudanças e seria necessário algumas migrações que não acho que vale apena resolver as dependências quebradas. 

Com isso em mente, pode surgir a possibilidade de trocar o Angular pelo React, mas isso depende muito da equipe, não sei como está o pessoal. Tem as vantagens e desvantagens nessa mudança, depende muito do perfil dos devs e de nossos planos para o futuro desse projeto.

**Comment by weslenbreno** [comment]
@rodrigondec Não manjo muito de BD mas vi que estávamos usando MySql, no entanto tenho percebido uma adoção maior do Postgres no mercado em relação ao Mysql. 

Estaremos na maior parte do tempo usando o ORM mesmo, no entanto, gostaria de saber quais as vantagens de a gente ficar com o MySql.

**Comment by rodrigondec** [comment]
@weslenbreno não me recordo da decisão do momento pra utilizar o MySQL tbm. Atualmente só trabalho com postgres, então não vejo motivo para não mudar tbm.

**Comment by gellyviana** [comment]
Sobre a mudança de Angular para React tenho algumas considerações a fazer, não tenho experiência com nenhuma das tecnologias, mas tenho pesquisado e o Angular é mais facil de aprender, entretanto no mercado, o React está sendo mais utilizado. Não sei se vocês conhecem, mas gostei muito desse podcast: https://hipsters.tech/angular-vs-react-hipsters-142/
Em todo caso, se fossemos fazer com React, precisaríamos fazer uma "capacitação" para que a comunidade possa contribuir.
Sobre a migração do repositório para o do DAAL, estarei falando com Lucas para agilizarmos essa parte.

**Comment by gellyviana** [comment]
Ah outra tecnologia concorrente do React é o Vue.js, @felihenrique poderia esclarecer melhor

**Comment by weslenbreno** [comment]
Conheço o **Vue.js**, acho excelente e fácil de desenvolver com ela. Mas ela ainda não é tão procurada no mercado. 

Penso que esse projeto pode capacitar a galera para uma futura vaga de emprego, seria ideal usar tecnologias que sejam boas e ainda tenha esse bônus. Por isso sugeri o React pois além da questão Web tem o React Native que ta sendo bem adotado pelo pessoal.

No entanto, eu me dou bem com as três não tenho problema na escolha.

**Comment by gellyviana** [comment]
Pessoal, quem poderia abrir o novo repositório no github do DAAL? 
Sobre o framework que iremos trabalhar no front, precisamos decidir o quanto antes, porque pretendo integrar a ideia do projeto na disciplina de Gerência e configuração de mudança, até para podermos avançar.

**Comment by rodrigondec** [comment]
Posso criar. Me adiciona como colaborador que eu subo

**Comment by gellyviana** [comment]
valeu

**Comment by DavidCardoso** [comment]
Também concordo com a mudança de frameworks. 
Em relação ao front, React tem mais peso no mercado. VueJS também está bem popular (tem até mais estrelas aqui no github do que o React). Contudo, o React tem mais contribuidores. Em suma, acho válido usar o projeto para capacitar a galera em um framework que poderá ajudar na entrada no mercado de trabalho (no caso, com React).

**Comment by gellyviana** [comment]
Muito obrigada pela contribuição. Então, vamos fechar? Com React para o front e Django para o Back?
 Já criamos o novo repositório no github do Daal e logo estaremos divulgando para a comunidade do BTI o nosso repositório aberto para contribuições. Quero saber se posso adicionar vcs no novo repositório? @weslenbreno, @Mazuh , @DavidCardoso e @sarahraquel

**Comment by Mazuh** [comment]
Pode sim, Gelly.

E concordo com David, inclusive estamos em busca de devs React (nível >=júnior) na Evolux há um tempo mas sem sucesso aqui em Natal.

**Comment by DavidCardoso** [comment]
Pode contar comigo, Gelly.

**Comment by gellyviana** [comment]
Adicionei vocês.

**Comment by weslenbreno** [comment]
Pode me adicionar, ficarei grato em contribuir com vocês.
